src/dodal/devices/apple2_undulator.py

```diff
@@ -69,10 +74,39 @@ async def estimate_motor_timeout(
     vel = await velocity.get_value()
     cur_pos = await curr_pos.get_value()
     target_pos = float(await setpoint.get_value())
-    return abs((target_pos - cur_pos) * 2.0 / vel) + DEFAULT_MOTOR_MIN_TIMEOUT
+    return abs((target_pos - cur_pos) * 2.0 / vel) + DEFAULT_TIMEOUT
+
+
+class UndulatorBase(abc.ABC, Device, Generic[T]):
+    """Abstract base class for Apple2 undulator devices.
+
+    This class provides common functionality for undulator devices, including
+    gate and fault signal management, safety checks before motion, and abstract
+    methods for setting demand positions and estimating move timeouts.
+    """
+
+    def __init__(self, name=""):
+        # Gate keeper open when move is requested, closed when move is completed
+        self.gate: SignalR[UndulatorGateStatus]
+        self.fault: SignalR[float]
+        super().__init__(name=name)
+
+    @abc.abstractmethod
+    async def _set_demand_positions(self, value: T) -> None:
+        """Set the demand positions on the device without actually hitting move."""
```

Comment:
```suggestion
class UndulatorBase(abc.ABC, Device, Movable[T], Generic[T]):
    """Abstract base class for Apple2 undulator devices.

    This class provides common functionality for undulator devices, including
    gate and fault signal management, safety checks before motion, and abstract
    methods for setting demand positions and estimating move timeouts.
    """

    def __init__(self, name: str = ""):
        # Gate keeper open when move is requested, closed when move is completed
        self.gate: SignalR[UndulatorGateStatus]
        self.fault: SignalR[float]
        super().__init__(name=name)

    @abc.abstractmethod
    async def _set_demand_positions(self, value: T) -> None:
        """Set the demand positions on the device without actually hitting move."""
```

nit: Can't this base class also be a Movable, if it's defining `_set_demand_positions`? It's an ABC, so it's not required to actually implement `set` yet.